numpy/core/shape_base.py

```diff
@@ -449,7 +450,7 @@ def _block_format_index(index):
     return 'arrays' + idx_str
 
 
-def _block_check_depths_match(arrays, parent_index=[]):
+def _block_check_depths_match(arrays, parent_index:List=None):
```

Comment:
A better pattern for avoiding this trap is to use `()` instead of `[]` and `mappingproxy()` instead of `{}`.

```suggestion
def _block_check_depths_match(arrays, parent_index=()):
```

This eliminates the need for `parent_index = parent_index or []` below.

Reply:
I don’t think this is better. The type of the same `parent_index ` will be uncertain, which will cause confusion for other modifiers.

Reply:
I'm suggesting that the best fix for this type of function is to change the type of the offending variable from `list` to `tuple`.

Reply:
That's right    ：D